### DISCUSSION TITLE
Many extensions missing?

### DISCUSSION THREAD

**Discussion by JJGadgets** [coder/code-server#2345] [Q&A]
I have installed code-server successfully on an Ubuntu AWS server, and am trying to sync my settings and extensions using Settings Sync. However, I am running into many errors generated by Settings Sync saying "Extension not found" for many of my extensions and have tried to remove them manually from the Gist created by Settings Sync, but out of 15 extensions so far, 7 are not found. I also cannot find the extensions by manually searching in the Code Server Extensions tab. Does Code Server use a different extensions database or is some other issue at play here?

**Comment by jaydns** [comment]
Yes, code server does use a different extension gallery. See https://github.com/Microsoft/vscode/issues/31168

**Reply by JJGadgets** [reply]
Is there a reason for this? I suppose it's due to compatibility issues? Is it okay to just change it back to the default once I've installed everything I want to use that's on the custom repo?

**Reply by jaydns** [reply]
> Is there a reason for this? I suppose it's due to compatibility issues? Is it okay to just change it back to the default once I've installed everything I want to use that's on the custom repo?

It's because Microsoft makes it unclear whether VSCode forks can use their extension gallery. You *can* switch to theirs, however it is discouraged as it might be violating their terms of use.

**Reply by oxy** [reply]
Microsoft does state that only their "Visual Studio" products are allowed to use their extension gallery in their [Terms of Use](https://aka.ms/vsmarketplace-ToU), under section 1b, Visual Studio Products and Services:

> Marketplace Offerings are intended for use only with Visual Studio Products and Services and you may only install and use Marketplace Offerings with Visual Studio Products and Services.

Hence, we maintain our own extension gallery, which may not have all the extensions that the VS Code gallery has. If you want to see an open source extension in the gallery, you can submit an extension request (by selecting the extension request option) [here](https://github.com/cdr/code-server/issues/new/choose).

**Reply by code-asher** [reply]
Update: we decommissioned our marketplace some time ago; code-server now uses Open VSX (https://open-vsx.org/) by default so if there are missing extensions please ask the extension authors to submit their extensions to Open VSX.

https://github.com/eclipse/openvsx/wiki/Publishing-Extensions
